crates/block-producer/src/withdrawal.rs

```diff
@@ -10,148 +9,209 @@ use gw_types::{
     bytes::Bytes,
     core::ScriptHashType,
     packed::{
-        CellDep, CellInput, CellOutput, CustodianLockArgs, DepositionLockArgs, GlobalState,
-        L2Block, RollupAction, RollupActionUnion, Script, ScriptOpt, Uint128,
-        UnlockWithdrawalViaRevert, UnlockWithdrawalWitness, UnlockWithdrawalWitnessUnion,
-        WithdrawalLockArgs, WithdrawalRequest, WitnessArgs,
+        CellDep, CellInput, CellOutput, CustodianLockArgs, DepositionLockArgs, L2Block,
+        RollupAction, RollupActionUnion, Script, UnlockWithdrawalViaRevert,
+        UnlockWithdrawalWitness, UnlockWithdrawalWitnessUnion, WithdrawalLockArgs,
+        WithdrawalRequest, WitnessArgs,
     },
     prelude::*,
 };
+use parking_lot::Mutex;
 
 use std::{
-    collections::HashSet,
+    collections::{HashMap, HashSet},
+    sync::Arc,
     time::{SystemTime, UNIX_EPOCH},
 };
 
-pub struct GeneratedWithdrawals {
-    pub deps: Vec<CellDep>,
-    pub inputs: Vec<InputCellInfo>,
-    pub outputs: Vec<(CellOutput, Bytes)>,
+#[derive(Debug)]
+pub struct AvailableCustodians {
+    pub capacity: u64,
```

Comment:
We should use u128 if it is an accumulated result.

Reply:
Done